f5router/bigipResources/bigipResources.go

```diff
@@ -148,18 +171,22 @@ type (
 	Rules    []*Rule
 	RouteMap map[route.Uri]*Pool
 	RuleMap  map[route.Uri]*Rule
-
-	// RoutingKey is the port for the route
-	RoutingKey struct {
-		Port uint16
-	}
-	// BackendServerKey is the endpoints info
-	BackendServerKey struct {
-		Address string
-		Port    uint16
-	}
 )
 
 func (r Rules) Len() int           { return len(r) }
 func (r Rules) Less(i, j int) bool { return r[i].FullURI < r[j].FullURI }
 func (r Rules) Swap(i, j int)      { r[i], r[j] = r[j], r[i] }
+
+func (va VirtualAddress) String() string {
+	return fmt.Sprintf("%s:%s", va.BindAddr, strconv.Itoa(int(va.Port)))
+}
+
+// Encode returns an encoded string of a VirtualAddress
+func (va VirtualAddress) Encode() string {
```

Comment:
Since some of your other changes added the pattern return val, nil | error in functions would that be worth adding here instead of just  returning the empty address string?

Reply:
Yup, updated to do so.